lib/oban/queue/executor.ex

```diff
@@ -70,8 +70,7 @@ defmodule Oban.Queue.Executor do
     Breaker.with_retry(fn -> report_finished(exec).state end)
   end
 
-  @spec record_started(t()) :: t()
-  def record_started(%__MODULE__{} = exec) do
+  defp record_started(%__MODULE__{} = exec) do
```

Comment:
These are public purposefully so that they can be composed arbitrarily by other modules. Let's keep this `def` instead.

```suggestion
  def record_started(%__MODULE__{} = exec) do
```